libs/blocks/event-rich-results/event-rich-results.js

```diff
@@ -0,0 +1,78 @@
+import { createTag } from '../../utils/utils.js';
+
+function buildData(el) {
+  const data = new Map();
+  const rows = el.children;
+  [...rows].forEach((row) => {
+    const key = [...row.children][0]?.innerText?.trim().replaceAll(' ', '-').toLowerCase();
+    const value = [...row.children][1]?.innerText;
+    data.set(key, value);
+  });
+  return data;
+}
+
+function getLocation(data) {
+  if (data.get('location-type') === 'Place') {
+    return {
+      '@type': 'Place',
+      name: data.get('location-name'),
+      address: {
+        '@type': 'PostalAddress',
+        streetAddress: data.get('location-address-street'),
+        addressLocality: data.get('location-address-locality'),
+        postalCode: data.get('location-address-postal-code'),
+        addressRegion: data.get('location-address-region'),
+        addressCountry: data.get('location-address-country'),
+      },
+    };
+  }
+  if (data.get('location-type') === 'VirtualLocation') {
+    return {
+      '@type': 'VirtualLocation',
+      url: data.get('location-url'),
+    };
+  }
+  return null;
```

Comment:
nit: could write a test for the null case from this function.

Reply:
@JasonHowellSlavin I updated the PR with a new test covering null/undefined properties at https://github.com/adobecom/milo/pull/511/commits/4136d46e818045fdc7fb50e45a57c1a9480f0b09